lib/install.js

```diff
@@ -79,277 +84,214 @@ function install (fs, gyp, argv, callback) {
-        }
+      log.verbose('checksum data', JSON.stringify(expectShasums))
+    }
 
-        req.on('error', done)
-        req.on('response', function (res) {
-          if (res.statusCode !== 200) {
-            done(new Error(res.statusCode + ' status code downloading checksum'))
-            return
+    async function downloadNodeLib () {
+      log.verbose('on Windows; need to download `' + release.name + '.lib`...')
+      const archs = ['ia32', 'x64', 'arm64']
+      return Promise.all(archs.map(async (arch) => {
+        const dir = path.resolve(devDir, arch)
+        const targetLibPath = path.resolve(dir, release.name + '.lib')
+        const { libUrl, libPath } = release[arch]
+        const name = `${arch} ${release.name}.lib`
+        log.verbose(name, 'dir', dir)
+        log.verbose(name, 'url', libUrl)
+
+        await fs.promises.mkdir(dir, { recursive: true })
+        log.verbose('streaming', name, 'to:', targetLibPath)
+
+        const res = await download(gyp, process.env, libUrl)
+
+        if (res.status === 403 || res.status === 404) {
+          if (arch === 'arm64') {
+            // Arm64 is a newer platform on Windows and not all node distributions provide it.
+            log.verbose(`${name} was not found in ${libUrl}`)
+          } else {
+            log.warn(`${name} was not found in ${libUrl}`)
           }
+          return
+        } else if (res.status !== 200) {
+          throw new Error(`${res.status} status code downloading ${name}`)
+        }
 
-          var chunks = []
-          res.on('data', function (chunk) {
-            chunks.push(chunk)
-          })
-          res.on('end', function () {
-            var lines = Buffer.concat(chunks).toString().trim().split('\n')
-            lines.forEach(function (line) {
-              var items = line.trim().split(/\s+/)
-              if (items.length !== 2) {
-                return
-              }
-
-              // 0035d18e2dcf9aad669b1c7c07319e17abfe3762  ./node-v0.11.4.tar.gz
-              var name = items[1].replace(/^\.\//, '')
-              expectShasums[name] = items[0]
-            })
-
-            log.verbose('checksum data', JSON.stringify(expectShasums))
-            done()
-          })
-        })
-      }
-
-      function downloadNodeLib (done) {
-        log.verbose('on Windows; need to download `' + release.name + '.lib`...')
-        var archs = ['ia32', 'x64', 'arm64']
-        var async = archs.length
-        archs.forEach(function (arch) {
-          var dir = path.resolve(devDir, arch)
-          var targetLibPath = path.resolve(dir, release.name + '.lib')
-          var libUrl = release[arch].libUrl
-          var libPath = release[arch].libPath
-          var name = arch + ' ' + release.name + '.lib'
-          log.verbose(name, 'dir', dir)
-          log.verbose(name, 'url', libUrl)
-
-          fs.mkdir(dir, { recursive: true }, function (err) {
-            if (err) {
-              return done(err)
```

Comment:
just noticing all of the potential double-callbacks for errors in this old code, yikes!